charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -217,24 +228,26 @@ def __init__(self, provider_charm, relation_name: str = DEFAULT_RELATION_NAME):
         self.provider_charm = provider_charm
         self.relation_name = relation_name
 
-    def send_gateway_relation_data(self, gateway_name: str, gateway_namespace: str) -> None:
-        """Updates the relation data bag with data from the localGateway.
+    def send_gateway_relation_data(self, gateway_name: str, gateway_namespace: str, gateway_up: bool = True) -> None:
+        """Updates the relation data bag of any related applications with data from the localGateway.
+
+        This method will complete successfully even if there are no related applications.
+
         Args:
             gateway_name (str): the name of the Gateway the provider knows about
             gateway_namespace(str): the namespace of the Gateway the provider knows about
+            gateway_up (bool): (optional) the status of the Gateway.  Assumed to be True by the
+                               requirer if not provided.
         """
         # Update the relation data bag with localgateway information
         relations = self.model.relations[self.relation_name]
 
-        # Raise if there is no related applicaton
-        if not relations:
-            raise GatewayRelationMissingError()
-
         # Update relation data
         for relation in relations:
             relation.data[self.provider_charm.app].update(
                 {
                     "gateway_name": gateway_name,
                     "gateway_namespace": gateway_namespace,
+                    "gateway_up": str(gateway_up).lower(),
```

Comment:
does not support bools or int ?

Reply:
no, the data bag is entirely string.  I was surprised too :p